bundles/persistence/org.openhab.persistence.sitewhere/src/main/java/org/openhab/persistence/sitewhere/internal/SiteWherePersistenceService.java

```diff
@@ -0,0 +1,443 @@
+			} catch (SiteWhereAgentException e) {
+				LOGGER.warn("Unable to store item: " + item.getName(), e);
+			}
+		}
+	}
+
+	/**
+	 * Send a device measurement to SiteWhere.
+	 * 
+	 * @param hardwareId
+	 * @param item
+	 * @param value
+	 * @throws SiteWhereAgentException
+	 */
+	protected void addDeviceMeasurement(String hardwareId, Item item,
+			double value) throws SiteWhereAgentException {
+		sitewhere.sendMeasurement(hardwareId, item.getName(), value, null);
+	}
+
+	/**
+	 * Send an alert for a {@link DateTimeType}.
+	 * 
+	 * @param hardwareId
+	 * @param item
+	 * @param value
+	 * @throws SiteWhereAgentException
+	 */
+	protected void addDeviceAlert(String hardwareId, Item item,
+			DateTimeType value) throws SiteWhereAgentException {
+		addDeviceAlert(hardwareId, item, TYPE_OPENHAB_DATETIME,
+				value.toString());
+	}
+
+	/**
+	 * Send an alert for an {@link OnOffType}.
+	 * 
+	 * @param hardwareId
+	 * @param item
+	 * @param value
+	 * @throws SiteWhereAgentException
+	 */
+	protected void addDeviceAlert(String hardwareId, Item item, OnOffType value)
+			throws SiteWhereAgentException {
+		addDeviceAlert(hardwareId, item, TYPE_OPENHAB_ONOFF, value.toString());
+	}
+
+	/**
+	 * Send an alert for an {@link OpenClosedType}.
+	 * 
+	 * @param hardwareId
+	 * @param item
+	 * @param value
+	 * @throws SiteWhereAgentException
+	 */
+	protected void addDeviceAlert(String hardwareId, Item item,
+			OpenClosedType value) throws SiteWhereAgentException {
+		addDeviceAlert(hardwareId, item, TYPE_OPENHAB_OPENCLOSED,
+				value.toString());
+	}
+
+	/**
+	 * Send an alert for a {@link StringType}.
+	 * 
+	 * @param hardwareId
+	 * @param item
+	 * @param value
+	 * @throws SiteWhereAgentException
+	 */
+	protected void addDeviceAlert(String hardwareId, Item item, StringType value)
+			throws SiteWhereAgentException {
+		addDeviceAlert(hardwareId, item, TYPE_OPENHAB_STRING, value.toString());
+	}
+
+	/**
+	 * Send a device alert to SiteWhere.
+	 * 
+	 * @param hardwareId
+	 * @param item
+	 * @param type
+	 * @param value
+	 * @throws SiteWhereAgentException
+	 */
+	protected void addDeviceAlert(String hardwareId, Item item, String type,
+			String value) throws SiteWhereAgentException {
+		String habType = type + ":" + item.getName();
+		sitewhere.sendAlert(hardwareId, habType, value, null);
+	}
+
+	/**
+	 * Add a device location to the given batch.
+	 * 
+	 * @param batch
+	 * @param info
+	 * @param point
+	 */
+	/*
+	 * protected void addDeviceLocation(DeviceEventBatch batch,
+	 * SiteWhereEventInfo info, PointType point) { DeviceLocationCreateRequest
+	 * loc = new DeviceLocationCreateRequest();
+	 * loc.setLatitude(point.getLatitude().doubleValue());
+	 * loc.setLongitude(point.getLongitude().doubleValue());
+	 * loc.setElevation(point.getAltitude().doubleValue());
+	 * loc.getMetadata().put("gravity", point.getGravity().toString());
+	 * loc.setEventDate(new Date()); loc.setUpdateState(true);
+	 * batch.getLocations().add(loc); }
+	 */
```

Comment:
why did you comment this code here? Could be interesting to track locations of e.g. the mover or house robot/pet which is connected to openHAB, isn't it?